src/main/kotlin/com/mdrobnak/lalrpop/inspections/PrecedenceInspection.kt

```diff
@@ -1,74 +1,82 @@
 
-                            if (level == null) {
-                                holder.registerProblem(arg.annotationArgValue, "Level value is not an int")
-                            }
-
-                            if (level == 1) {
-                                if (assocAnnotation != null) {
-                                    holder.registerProblem(
-                                        assocAnnotation,
-                                        "#[assoc] used on alternative with #[precedence(level=\"1\")]"
-                                    )
-                                }
-                            }
-                        }
+                    if (level == null) {
+                        holder.registerProblem(arg.annotationArgValue, "Level value is not an int")
                     }
 
-
-                    if (assocAnnotation != null) {
-                        val assocAnnotationArg = assocAnnotation.annotationArg
-
-                        if (assocAnnotationArg == null || assocAnnotationArg.annotationArgName.text != "side") {
-                            holder.registerProblem(assocAnnotation, "Missing side=\"...\" on #[assoc] annotation")
-                        }
+                    if (level == 1 && assocAnnotation != null) {
+                        holder.registerProblem(
+                            assocAnnotation,
+                            """#[assoc] used on alternative with #[precedence(level="1")]"""
+                        )
                     }
+                }
+            }
 
-                    if (precedenceAnnotation == null) {
-                        alternativesWithoutPrecedenceAnnotation.add(it)
+            if (assocAnnotation != null) {
+                assocAnnotation.annotationArg.takeUnless { it != null && it.annotationArgName.text == "side" }
+                    ?.let {
+                        holder.registerProblem(assocAnnotation, """Missing side="..." on #[assoc] annotation""")
                     }
+
+                if (precedenceAnnotationNullable == null) {
+                    holder.registerProblem(
+                        assocAnnotation,
+                        """#[assoc] annotation without #[precedence] annotation""",
+                        DeleteAnnotationQuickFix("assoc")
+                    )
                 }
+            }
+        }
 
-                if (hasPrecedence) {
-                    alternativesWithoutPrecedenceAnnotation.forEach { alternative ->
-                        holder.registerProblem(
-                            alternative,
-                            "Alternative without #[precedence] in a nonterminal that has a #[precedence] alternative"
-                        )
-                    }
+        override fun visitAlternatives(alternatives: LpAlternatives) {
+            val (withPrecedence, withoutPrecedence) = alternatives.alternativeList.partition { it.findAnnotationByName("precedence") != null }
```

Comment:
This line is way too long.
```suggestion
            val (withPrecedence, withoutPrecedence) = alternatives.alternativeList
                .partition { it.findAnnotationByName("precedence") != null }
```